Docs: New panel query editor public preview (#126590)
grafana/grafana: docs/sources/visualizations/panels-visualizations/query-transform-data/_index.md

title: Query and transform data
description: Query and transform your data
weight: 40
keywords:
  - saved queries
  - reuse queries
  - saved query
  - reuse query
image_maps:
  - key: query-editor
    src: query-editor-temp.png
    alt: An annotated image of the Grafana panel query editor
    points:
      - x_coord: 1
        y_coord: 50
        content: |
          **Sidebar**

          The sidebar displays queries, expressions, and transformations as color-coded cards with visual indicators for state, like error, hidden, and disabled, while a footer tracks your total item count.
      - x_coord: 13
        y_coord: 8
        content: |
          **Data/Alerts buttons**

          Click the buttons to move between data pipeline and related alerts. In **Alerts**, you have the option to start creating a alert rule.
      - x_coord: 24
        y_coord: 8
        content: |
          **Stacked view icon**

          The [stacked view](#navigate-the-query-editor) displays all of your queries, expressions, and transformations in a single list in the editor pane.
      - x_coord: 1
        y_coord: 1
        content: |
          **Multi-select**

>           Click **Select...** in the sidebar footer to enter multi-select mode, then check the items you want to work with so you can take bulk actions. You can also press the `Shift` key while you click to select a range. If you select queries that all have the same data source, you can change the data source for all of them at once.

Reviewer: This is a little bit off I think! 

```suggestion
          Click **Select...** in the sidebar footer to enter multi-select mode, then check the items you want to work with so you can take bulk actions. You can also press the `Shift` key while you click to select a range. If all items in your selection are queries, you can change the data source for all of them at once.
```

To use this feature, enable the `queryEditorNext` feature toggle in your Grafana configuration file or contact Support.
{{< /admonition >}}

Each data source's query editor provides a customized user interface that helps you write queries that take advantage of its unique capabilities.

Because of the differences between query languages, each data source query editor looks and functions differently.
Depending on your data source, the query editor might provide auto-completion features, metric names, variable suggestions, or a visual query-building interface.

{{< figure src="/media/docs/grafana/panels-visualizations/screenshot-queries-tab-v11.6.png" max-width="750px" alt="The InfluxDB query editor" >}}

<!-- TODO: Update screenshot but I can't get this to render properly -->

For details on a specific data source's unique query editor features, refer to its documentation:

- For data sources included with Grafana, refer to [Built-in core data sources](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/datasources/#built-in-core-data-sources), which links to each core data source's documentation.
- For data sources installed as plugins, refer to the documentation for the plugin.

>   - Data source plugins in Grafana's [plugin catalog](/grafana/plugins/) link to or include their documentation in their catalog listings.

Author: ```suggestion
  - Data source plugins in the Grafana [plugin catalog](/grafana/plugins/) link to or include their documentation in their catalog listings.
```
